pandas/core/dtypes/cast.py

```diff
@@ -1841,8 +1841,10 @@ def maybe_cast_to_integer_array(
             f"casted to the dtype {dtype}"
         ) from err
 
-    if np.array_equal(arr, casted):
-        return casted
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore")
```

Comment:
Curious what warnings we're filtering here and should we make the filter more specific?

Reply:
Hm not sure if regarding more specific.

We are filtering a FutureWarning regarding elementwise comparisons that's hidden in there